### PR TITLE
Indexer ValidateAndWaitOnly startup mode for canton participant HA

### PR DESCRIPTION
**Pull request by oliverse-da**
Port of change from canton: https://github.com/DACH-NY/canton/pull/6556 but avoids reuse of existing indexer startup mode.

Implementation per "Design: Indexer HA Support for Canton" doc. - fyi @dasormeter , @nmarton-da , @mziolekda , @soren-da

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [X] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala [hunk 1]
```diff
@@ -68,6 +69,49 @@ private[platform] class FlywayMigrations(jdbcUrl: String)(implicit loggingContex
...
+      val flyway = configurationBase(dbType, enableAppendOnlySchema)
+        .dataSource(ds)
+        .ignoreFutureMigrations(false)
+        .load()
+      logger.info("Running Flyway validation...")
+
+      @tailrec
+      def flywayMigrationDone(
+          retries: Int,
+          needLessThan: Option[Int],
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala:86]: Can you think of a better name? I found myself having to go to the implementation to figure out what this argument is about, which is not a good sign.

**Comment by mziolekda** [comment]
There is one thing I don't understand about this PR. In the design document we write:

> Introduce IndexerStartupMode.ValidateAndWaitOnly
> Flyway: introducing a new method for retrying validation (see Oliver Seeliger’s suggestion here)
> **Ledger initialization: no**
> Start indexer: no
> Canton configures: for passive nodes
the `initalized` function is still being called however, meaning that the ledger will be initialized

File: ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala (hunk 1 above)
Comment (oliverse-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala]: Sure, changing `needLessThan` to be more "declarative" `pendingMigrationsSoFar`.

**Comment by oliverse-da** [comment]
> There is one thing I don't understand about this PR. In the design document we write:
> 
> > Introduce IndexerStartupMode.ValidateAndWaitOnly
> > Flyway: introducing a new method for retrying validation (see Oliver Seeliger’s suggestion here)
> > **Ledger initialization: no**
> > Start indexer: no
> > Canton configures: for passive nodes
> 
> the `initalized` function is still being called however, meaning that the ledger will be initialized

Good catch - removed.

**Review by mziolekda — APPROVED** [review]
Looks good now

**Pull request merged**